meta-mender-core/classes/mender-part-images.bbclass

```diff
@@ -124,9 +124,11 @@ EOF
         bbwarn "MENDER_BOOT_PART_SIZE_MB is set to zero, but IMAGE_BOOT_FILES is not empty. The files are being omitted from the image."
     fi
 
-    # 'squashfs' fstype doesn't support empty partitions. For all others we make
-    # it empty to save compression space.
-    if [ "${ARTIFACTIMG_FSTYPE}" = "squashfs" ]; then
+    # By default the inactive partition filesystem is empty to allow for more efficient compression.
+    # A full filesystem is populated if one of the following applies
+    # - ARTIFACTIMG_FSTYPE is squashfs, because it doesn not allow empty partitions.
+    # - the "mender-prepopulate-inactive-partition" MENDER_FEATURE is enabled
+    if [ "${ARTIFACTIMG_FSTYPE}" = "squashfs" || ${@bb.utils.contains('MENDER_FEATURES', 'mender-prepopulate-inactive-partition', 'true', 'false', d)} ]; then
```

Comment:
Any thoughts about a warning in case for the squashfs case without the `mender-prepopulate-inactive-partition`?

Something like "squashfs detected, the build will behave as if `mender-prepopulate-inactive-partition` feature is enabled"

It makes things more explicit while at the same time not breaking anything already present.

Reply:
@TheMeaningfulEngineer technically correct, but it changes the existing behavior of `ARTIFACTIMAGE_FSTYPE = "squashfs"` and therefore it should be a second commit in my opinion. Read: I thought about it, but decided against mangling it into this commit, also as there are several approaches to do it and I don't know yet which is the best.

Reply:
Commit or PR?